### PR TITLE
GEOMESA-2913 Support Kryo deserialization of old geotools hint keys

### PR DESCRIPTION
**Pull request by michaelchehreh-ccri**
Signed-off-by: Michael Chehreh <mchehreh@ccri.com>

### CODE REVIEW

File: geomesa-features/geomesa-feature-kryo/src/main/scala/org/locationtech/geomesa/features/kryo/serialization/KryoUserDataSerialization.scala [hunk 1]
```diff
@@ -43,7 +43,12 @@ object KryoUserDataSerialization extends LazyLogging {
     classOf[Hints.Key]         -> "$h"
   )
 
-  private val baseClassLookups: Map[String, Class[_]] = baseClassMappings.filterNot(_._1.isPrimitive).map(_.swap)
+  private val baseClassLookups: Map[String, Class[_]] = {
+    val m1 = baseClassMappings.filterNot(_._1.isPrimitive).map(_.swap)
+    // support hints generated with geotools versions <= 20
+    val m2 = m1 + ("org.geotools.factory.Hints$Key" -> classOf[Hints.Key])
+    m2
+  }
```
Comment (michaelchehreh-ccri) [inline geomesa-features/geomesa-feature-kryo/src/main/scala/org/locationtech/geomesa/features/kryo/serialization/KryoUserDataSerialization.scala:51]: @elahrvivaz feel free to rework this w/ better syntax/comments/etc.

Comment (michaelchehreh-ccri) [inline geomesa-features/geomesa-feature-kryo/src/main/scala/org/locationtech/geomesa/features/kryo/serialization/KryoUserDataSerialization.scala:51]: @elahrvivaz feel free to rework this w/ better syntax/comments/etc.

Deleted the old comment and created a new one w/ the multi-line-select feature. Fun :)

**Review by elahrvivaz — APPROVED** [review]
lgtm!

File: geomesa-features/geomesa-feature-kryo/src/main/scala/org/locationtech/geomesa/features/kryo/serialization/KryoUserDataSerialization.scala (hunk 1 above)
Comment (elahrvivaz) [reply geomesa-features/geomesa-feature-kryo/src/main/scala/org/locationtech/geomesa/features/kryo/serialization/KryoUserDataSerialization.scala:51]: looks good as is!

**Pull request merged**